src/libServer/Server.cpp

```diff
@@ -1376,4 +1377,61 @@ string Server::GetNodeState() {
   } else {
     return m_mediator.m_ds->GetStateString();
   }
+}
+
+bool Server::AddToBlacklistExclusion(const string& ipAddr) {
+  if (LOOKUP_NODE_MODE) {
+    throw JsonRpcException(RPC_INVALID_REQUEST, "Not to be queried on lookup");
+  }
+
+  try {
+    boost::multiprecision::uint128_t numIP;
+
+    if (!IPConverter::ToNumericalIPFromStr(ipAddr, numIP)) {
+      throw JsonRpcException(RPC_INVALID_PARAMETER,
+                             "IP Address provided not valid");
+    }
+
+    if (!Blacklist::GetInstance().Exclude(numIP)) {
+      throw JsonRpcException(
+          RPC_INVALID_PARAMETER,
+          "Could not add IP Address in exclusion list, already present");
```

Comment:
There is another case. i.e blacklist is disabled. 
Maybe good to refine the message